src/lib/utils.ts

```diff
@@ -1,46 +1,61 @@
 import clsx, { type ClassValue } from "clsx";
 import { twMerge } from "tailwind-merge";
+import { z } from "zod";
 
 export const cn = (...inputs: ClassValue[]) => {
   return twMerge(clsx(inputs));
 };
 
-interface Release {
-  tag_name: string;
-}
-
-export interface VersionsGroupedByMajor {
-  [major: string]: string[];
-}
+export type VersionsGroupedByMajor = Array<{
+  major: string;
+  versions: string[];
+}>;
 
 export const getT3Versions = async () => {
   const response = await fetch(
     "https://api.github.com/repos/t3-oss/create-t3-app/releases"
   );
-  const data = (await response.json()) as Release[];
-  const versions = data.map((release) => release.tag_name.split("@")[1] ?? "");
-  const actualVersions = versions.filter((version) => version !== "");
 
-  return actualVersions;
+  const regexForVersion = /^create-t3-app@\d+\.\d+\.\d+$/;
+
+  const responseSchema = z.array(
+    z.object({ tag_name: z.string().regex(regexForVersion) })
+  );
+  const parsed = responseSchema.safeParse(await response.json());
+
+  if (!parsed.success) {
+    return [];
+  }
```

Comment:
Should we filter out the things that don't match rather than doing an early return with [] ? Just in case something else gets published under create-t3-app repo ? (really unlikely I think but just to be safe)

Reply:
Good point, one bad value would make everything fail. Fixing now

Reply:
And fixed :)